flaskext/kvsession.py

```diff
@@ -137,9 +137,16 @@ def open_session(self, app, request):
 
                     ## Kenji
                     # retrieve from store
-                    s = KVSession(self.serialization_method.loads(
-                        current_app.kvsession_store.get(sid_s))
-                    )
+                    try:
+                        s = KVSession(self.serialization_method.loads(
+                            current_app.kvsession_store.get(sid_s))
+                        )
+                    except (UnicodeDecodeError, ValueError):
+                        # If a cookie is reused between different app versions (Python 2 and Python 3), one of the
+                        # above exceptions will be raised. Just delete the associated session and create a new one
+                        current_app.kvsession_store.delete(sid_s)
```

Comment:
DON’T delete sessions, but restore session from python2 serialization format and convert to
python3 serialization format.

Reply:
another option: always use pickle protocol version 2

Reply:
Yesterday, I first took the approach you mentioned above and found it worked. However, I encountered another serialisation error when downgrading the Python version from 3 to 2. I mean, the error occurred when unpickling an object which was pickled from Python 3 previously.  I'm not sure that we could fix the error in this case. So, I decided to take the DELETE-SESSION approach.

Reply:
>Use protocol 2
Okay. Hope that there were no sessions created during the deployment several days ago :D

Reply:
I just updated the code. Please check it out.

Scenarios I have checked:
- Reusing a session which was created from Python 2 app version after upgrading to Python 3
- Reusing a session which was created from Python 3 app version (serialised with protocol 2) after downgrading to Python 2
- Reusing a session which was created from Python 3 app version (serialised with protocol 4) after downgrading to Python 2 => This is for sessions generated during our deployment on 15 Jan 2024. I'd like to keep deleting sessions in this case